viewing_party/party.py

```diff
@@ -1,23 +1,312 @@
+                    # print(watched_dict["title"])
+                    if watched_dict not in friends_watched_list:
+                        friends_watched_list.append(watched_dict)
         
+        return friends_watched_list
+
+
+def get_user_watched_list(user_data):
+    """Helper function to create user's watch list
+    Input: user data
+    Output: list of user's movie dictionaries
+    
+    """
+    user_watched_list = []
+
+    if len(user_data["friends"]) ==0:
+        user_watched_list = []
+    else:
+        for movie in range(len(user_data["watched"])):
+            user_watched_list.append(user_data["watched"][movie])
+    
+    return user_watched_list
+           
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+def get_available_recs(user_data):
+    to_remove_list = get_to_remove_list(user_data)
+
+    user_recs = get_friends_unique_watched(user_data)
+    for element in user_recs[:]:
+        if element in to_remove_list:
+            user_recs.remove(element)
+    return user_recs
+
+
+def get_to_remove_list(user_data):
+    """Helper function to get list of 'subscription' service user do not have
+    Input: user_data
+    Output: list of movie to remove
+    
+    """    
+    friends_unique_list = get_friends_unique_watched(user_data)
+    user_recs = friends_unique_list.copy()
+    to_remove_list = []
+    user_subscription_list = user_data["subscriptions"]
+    for element in range(len(user_recs)):
+        if (user_recs[element]["host"]) not in user_subscription_list:
+            to_remove_list.append(user_recs[element])
+    # print(to_remove_list)
+    return to_remove_list
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+
+def get_new_rec_by_genre(user_data):
+    user_most_popular_genre = get_most_watched_genre(user_data) #get_pop_genre(user_data) 
+    
+    user_recs = get_friends_unique_watched(user_data)
+    recommendations = []
+    
+    for movie in user_recs:
+        if movie["genre"] == user_most_popular_genre:
+            recommendations.append(movie)
+    print(recommendations)
+    return recommendations
+
+def get_rec_from_favorites(user_data):
+    
+    user_favorite_list = get_user_fav(user_data)
+    to_rec_to_friend = get_unique_watched(user_data)
+    if len(user_data["friends"]) == 0:
+        to_rec_to_friend = user_favorite_list
+        print("MOVIE RECS ARE",to_rec_to_friend)
+        return to_rec_to_friend
+    else:
+        for movie in to_rec_to_friend:
+            print("movie to recomend are",movie)
+            if movie not in user_favorite_list:
+                to_rec_to_friend.remove(movie)
+        print("MOVIE RECS",to_rec_to_friend)
+        return to_rec_to_friend
+    
+    
+def get_user_fav(user_data):

```

Comment:
Again, great instinct to use a helper function, but this functionality also works on one line: `user_favorite = user_data["favorite"]`